### DISCUSSION TITLE
LO/TO spiliting discussion question

### DISCUSSION THREAD

**Discussion by Seonghyeonkweon** [ttadano/alamode#100] [Q&A]
hello ttadono
First of all, thank you for providing a good phonon calculation system.

I have an additional question about your answer on how to distinguish phonon mode (LO/TO/ZO...).
https://github.com/ttadano/alamode/discussions/97

I want to calculate temperature dependent dielectric constant of BaTiO3 perovskite through scph calculation.
To calculate the high-temperature dielectric constant, the gamma point frequency at that temperature is needed.

Through spch calculation, I found 15 phonon modes at the gamma point (.scph_band file), and I want to classify these 15 modes as LA/TA/ZA/LO/TO/ZO.

In the last question you gave 2 comments.

1. Compute the dot products (and cross products) of the phonon polarization vector and the momentum vector
-> I think this solution is the best.
However, I couldn't find any information about the phonon polarization vector and momentum vector in the result file of my scph calculation. So, how can I get the phonon polarization vector and momentum vector from the scph calculation result? 

2. Compare phonon dispersion curves obtained using NONANALLYTIC=0 and NONANALYTIC=3, you will see differences in the frequencies of LO phonons.
-> Since NONANLLYTIC=3 does not apply to the scph calculation, I tried to compare the calculation results of NONANLLYTIC=0 and NONANLLYTIC=2.

![https://user-images.githubusercontent.com/115207966/196952831-019c5aa7-2d39-4f90-a3e3-2f99804cdf14.png](https://user-images.githubusercontent.com/115207966/196952831-019c5aa7-2d39-4f90-a3e3-2f99804cdf14.png)

-> this graph is scph calculation result with (NONANLLYTIC=0)

![https://user-images.githubusercontent.com/115207966/196952769-43f80e16-b199-41f7-ae7f-2cbe2b8f6043.png](https://user-images.githubusercontent.com/115207966/196952769-43f80e16-b199-41f7-ae7f-2cbe2b8f6043.png)

-> this graph is scph calculation result with (NONANLLYTIC=2)

As a result of comparing the frequencies of the two results, there was a difference in frequency only at 0K and 50K, and it was completely the same at 100K or higher.

Even in the 0K and 50K results, all the frequencies of the gamma point were different, so it was impossible to distinguish the LO.

Therefore, I am trying to apply the first method (phonon polarization vector and momentum vector) that you pointed out, and I desperately need help from developers.

**Comment by ttadano** [comment]
The second figure looks strange because LO-TO splitting is absent. How did you get this figure?
The expected procedure is as follows:

1. Run SCPH with `NONANALYTIC=2` (or `NONANALYTIC=0`) and save PREFIX.scph_dymat
2. Rerun SCPH with `NONANALYTIC=3` as a restart mode (`RESTART_SCPH=1` automatically set when PREFIX.scph_dymat exists)

About the first question, the SCPH mode (`MODE = SCPH`) cannot print eigenvectors to a file. So, please do the following:

1. Run SCPH and save PREFIX.scph_dfc2
2. Run `dfc2` program in ${ALAMODE_ROOT}/tools/ directory as 
`> dfc2 original_fc2.xml renormalized_fc2_500K.xml prefix.scph_dfc2 500`
 The above command saves effective 2nd-order force constants at 500 K in `renormalized_fc2_500K.xml`
3. Run harmonic phonon calculation (`MODE = phonons`) with `FCSXML = renormalized_fc2_500K.xml` and `PRINTEVEC = 1`. If you want, you can set `NONANALYTIC=3` (1 and 2 are fine as well) to include the LO-TO splitting.
